src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        if (quantity < 0) {
+            throw new IllegalArgumentException("Quantity can't be negative");
+        }
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getOperationByCode(String operationCode) {
+            for (Operation operation : Operation.values()) {
+                if (operation.getCode().equals(operationCode)) {
+                    return operation;
+                }
+            }
+            return null;
```

Comment:
It will be cool if you throw new NoSuchElementException when we don't have correct code.